vmrunner/vmrunner.py

```diff
@@ -516,6 +519,41 @@ def net_arg(self, backend, device, if_name = "net0", mac = None, bridge = None,
         return ["-device", device,
                 "-netdev", netdev]
 
+    def init_virtiocon(self, path):
+        """ creates a console device and redirects to the path given """
+        qemu_args = ["-device", "virtio-serial-pci,disable-legacy=on,id=virtio-serial0"]
+        qemu_args += ["-device", "virtserialport,chardev=virtiocon0"]
+        qemu_args += ["-chardev", f"file,id=virtiocon0,path={path}"]
+
+        return qemu_args
+
+    def init_virtiofs(self, socket, shared, mem):
+        """ initializes virtiofs by launching virtiofsd and creating a virtiofs device """
+        virtiofsd_args = ["virtiofsd", "--socket", socket, "--shared-dir", shared, "--sandbox", "none"]
+        self._virtiofsd_proc = subprocess.Popen(virtiofsd_args, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL) # pylint: disable=consider-using-with
+
+        if self._virtiofsd_proc.poll():
+            raise Exception("VirtioFSD failed to start")
+
+        info("Successfully started VirtioFSD!")
+
+        while not os.path.exists(socket):
```

Comment:
This will hang with no helpful information if the shared folder doesn't exist: 
```
* <VMRunner>:  Booting ./fs_virtiopmem.elf.bin directly without bootloader (multiboot / -kernel args) 
* <VMRunner>:  Successfully started VirtioFSD! 

```
You can reproduce this by adding this to vm.json
```
  "virtiofs" : {
      "shared" : "./this_is_shared"
      
  }
```

And then booting with `boot`. After creating the directory `this_is_shared`, the unikernel starts. So we should check if the provided shared folder exists and is a folder, and if not error out. But since it works in the happy path, and only takes effect if you actually add a `virtiofs` entry to `vm.json`, I'm happy to merge this - just add the check in a separate PR.